### ISSUE TITLE
Amazon Athena: Variable 'result re-use' not working

### ISSUE THREAD

**Issue by MatinF** [grafana/athena-datasource#302]
### What happened?

I'm aiming to use 'result reuse' in my Amazon Athena based dashboard across all panels and also in my Variables.

I have set this up with the result reuse enabled on all my Variables. However, when refreshing the dashboard with the exact same variable values selected (incl. generic statements like `SHOW TABLES`), the results are in fact not re-used as evident from the Athena queries overview. This causes everything to slow down significantly.

I've shown this in the below video:

https://github.com/grafana/grafana/assets/26184427/103d6667-494b-4d24-b772-db3c87776a6d

You can access the dashboard from the video below if it helps:
[dashboard playground](https://grafana.csselectronics.stellarhosted.com/d/obd2-gps-car/obd2-gps-car?orgId=1)

### What did you expect to happen?

I expect that Variable Athena queries support result reuse

### Did this work before?

I am not sure if it has worked before

### How do we reproduce it?

See the video and playground.

### Environment (with versions)?

Grafana: Cloud
OS: Windows
Browser: Chrome

### Grafana platform?

I use Grafana Cloud

**Comment by idastambuk** [comment]
Hi @MatinF I have transferred this issue into the repo for the Athena plugin, where we keep bugs and issues for the plugin. Looking into your problem, will let you know when we have updates

**Comment by idastambuk** [comment]
Hi @MatinF thanks for giving us access to the dashboard! From my initial investigation, I think there are two different issues here: 
1. Your `SELECT 
    array_join(...` query that isn't getting cached does not have Result reuse enabled in the variable query settings. Can you enable it and see if that fixes it for that specific query? 
2. The SHOW queries are indeed not getting cached in Athena. However, when I test those queries in my AWS Athena console, they are not getting cached there either (or, at least not being flagged as "Cached"). This leads me to believe this is not a Grafana Athena plugin issue, but an unexpected behavior in AWS itself. 
My suggestion would be for you to open an support ticket in AWS to check why these queries aren't getting cached.

**Comment by MatinF** [comment]
I have tried to enable 'result reuse' on the `SELECT array_join(...` as well, but it does not change the result:

![image](https://github.com/grafana/athena-datasource/assets/26184427/f8a72942-f521-48f3-856a-37ae5db84d0f)

I also tried using the same query in a panel rather than a variable, but the result remains the same - for some reason, this particular SELECT query fails to be reused, which is unfortunate as it's causing significant slowdown in our dashboards. 

Appreciate the tip on AWS, though I don't think they'll do anything about it if the request comes from our side - though they might if it comes from your team. Either way, appreciate the support on this!

**Comment by sarahzinger** [comment]
@MatinF If the results are the same as what you're seeing in the AWS console, I suspect there's not much we can do on our end. We've reached out to our contacts at AWS to let them know, but your best bet is probably to open a support ticket with them.

Some additional ideas to help in the meantime:
- I was looking through AWS's docs on the subject and they had a list of potential limitations that it might be good to run through: https://docs.aws.amazon.com/athena/latest/ug/reusing-query-results.html#reusing-query-results-considerations-and-limitations one thing that stood out to me was that it can only reuse deterministic results. I wonder for example if adding an `order by` could help in some way with the Select query?
- You mentioned you're using Grafana Cloud and there is a feature in Grafana Cloud called [Query Caching](https://grafana.com/blog/2021/09/02/reduce-costs-and-increase-performance-with-query-caching-in-grafana-cloud/) and last I checked Athena supports it as well. Could be interesting to see if that helps performance a bit by enabling that setting in your Datasource Configuration settings.
